common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -750,6 +753,81 @@ public AccountDeletionRecord removeAccount(@Nullable final String environment,
         );
     }
 
+    /**
+     * Tests if a clientId is 'known' to the cache. A clientId is known if a token has been
+     * previously saved to the cache with it.
+     *
+     * @param clientId The clientId whose known status should be queried.
+     * @return True if this clientId is known. False otherwise.
+     */
+    public boolean isClientIdKnownToCache(@NonNull final String clientId) {
+        return getAllClientIds().contains(clientId);
+    }
+
+    /**
+     * Returns the List of FoCI users in the cache. This API is provided so that the broker may
+     * **internally** query the cache for known users, such that the broker may verify an
+     * unknown clientId is a part of the FoCI family.
+     * <p>
+     * Please note, the ICacheRecords returned by this query are NOT fully populated. Only the
+     * {@link GenericAccount} and {@link GenericRefreshToken} will be returned.
+     * will be resutned.
+     *
+     * @param environment The environment whose accounts should be queried.
+     * @return A List of ICacheRecords for the FoCI accounts.
+     */
+    @SuppressWarnings("unchecked")
+    public List<ICacheRecord> getFociCacheRecords(@NonNull final String environment) {
+        final List<ICacheRecord> fociCacheRecords = new ArrayList<>();
+
+        final List<Pair<AccountRecord, String>> fociAccounts = new ArrayList<>();
+
+        final Set<String> fociClientIds = mApplicationMetadataCache.getAllFociClientIds();
+
+        for (final String fociClientId : fociClientIds) {
+            // Get the list of all of the accounts for this client...
+            final List<AccountRecord> clientAccounts = mFociCache.getAccounts(
+                    environment,
+                    fociClientId
+            );
+
+            // Add them as Pairs...
+            for (final AccountRecord record : clientAccounts) {
+                fociAccounts.add(
+                        new Pair<>(
+                                record,
+                                fociClientId
+                        )
+                );
+            }
+        }
+
+        // For each account, load its refresh token and add it to the result List...
+        for (final Pair<AccountRecord, String> recordPair : fociAccounts) {
+            // Load the refresh token for this account...
+            final List<Credential> refreshTokens = mFociCache
+                    .getAccountCredentialCache()
+                    .getCredentialsFilteredBy(
+                            recordPair.first.getHomeAccountId(),
+                            environment,
```

Comment:
Checked `SharedPreferencesAccountCredentialCache#getCredentialsFilteredBy`, the `environment` here is also nullable.